vgscollect/src/main/java/com/verygoodsecurity/vgscollect/view/internal/CVCInputField.kt

```diff
@@ -155,25 +138,29 @@ internal class CVCInputField(context: Context) : BaseInputField(context) {
         this.previewIconGravity = PreviewIconGravity.values()[gravity]
     }
 
+    internal fun setPreviewIconAdapter(adapter: CvcIconAdapter?) {
+        this.iconAdapter = adapter ?: CvcIconAdapter(context)
+    }
+
     private fun refreshIcon() {
         when (previewIconVisibility) {
-            ALWAYS -> setIcon(getIcon())
-            HAS_CONTENT -> setIcon(if (text.isNullOrEmpty()) null else getIcon())
-            IF_BRAND_DETECTED -> setIcon(if (cardType == CardType.UNKNOWN) null else getIcon())
-            NEVER -> setIcon(null)
+            ALWAYS -> setIcon()
+            HAS_CONTENT -> if (text.isNullOrEmpty()) removeIcon() else setIcon()
```

Comment:
show\hide , remove\add, etc..

Reply:
@Endorf Actually in this case `add` will no be good, because `add` mean that the prev item will not disappear because you `add` it, but when you `set` something it's mean you override it. Show/hide I think in this case also not showing real picture of what is going on there.